src/Core/Domain/CartRule/QueryResult/EditableCartRule.php

```diff
@@ -0,0 +1,557 @@
+
+    /**
+     * @var int
+     */
+    private $priority;
+
+    /**
+     * @var bool
+     */
+    private $partialUse;
+
+    /**
+     * @var string
+     */
+    private $code;
+
+    /**
+     * @var Number
+     */
+    private $minimumAmount;
+
+    /**
+     * @var bool
+     */
+    private $minimumAmountTax;
+
+    /**
+     * @var CurrencyId
+     */
+    private $minimumAmountCurrencyId;
+
+    /**
+     * @var bool
+     */
+    private $minimumAmountShipping;
+
+    /**
+     * @var bool
+     */
+    private $countryRestriction;
+
+    /**
+     * @var bool
+     */
+    private $carrierRestriction;
+
+    /**
+     * @var bool
+     */
+    private $groupRestriction;
+
+    /**
+     * @var bool
+     */
+    private $cartRuleRestriction;
+
+    /**
+     * @var bool
+     */
+    private $productRestriction;
+
+    /**
+     * @var bool
+     */
+    private $shopRestriction;
+
+    /**
+     * @var bool
+     */
+    private $freeShipping;
+
+    /**
+     * @var Number
+     */
+    private $reductionPercent;
+
+    /**
+     * @var Number
+     */
+    private $reductionAmount;
+
+    /**
+     * @var bool
+     */
+    private $reductionTax;
+
+    /**
+     * @var CurrencyId
+     */
+    private $reductionCurrencyId;
+
+    /**
+     * @var ProductId
+     */
+    private $reductionProductId;
+
+    /**
+     * @var bool
+     */
+    private $reductionExcludeSpecial;
+
+    /**
+     * @var ProductId
+     */
+    private $giftProductId;
+
+    /**
+     * @var CombinationId
+     */
+    private $giftCombinationId;
+
+    /**
+     * @var bool
+     */
+    private $highlight;
+
+    /**
+     * @var bool
+     */
+    private $enabled;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateAdd;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateUpd;
+
+    public function __construct(
+        CartRuleId $cartRuleId,
+        array $localizedNames,
+        ?CustomerId $customerId,
+        ?DateTime $dateFrom,
+        ?DateTime $dateTo,
+        string $description,
+        int $quantity,
+        int $quantityPerUser,
+        int $priority,
+        bool $partialUse,
+        string $code,
+        Number $minimumAmount,
+        bool $minimumAmountTax,
+        CurrencyId $minimumAmountCurrencyId,
+        bool $minimumAmountShipping,
+        bool $countryRestriction,
+        bool $carrierRestriction,
+        bool $groupRestriction,
+        bool $cartRuleRestriction,
+        bool $productRestriction,
+        bool $shopRestriction,
+        bool $freeShipping,
+        Number $reductionPercent,
+        Number $reductionAmount,
+        bool $reductionTax,
+        ?CurrencyId $reductionCurrencyId,
+        ?ProductId $reductionProductId,
+        bool $reductionExcludeSpecial,
+        ?ProductId $giftProductId,
+        ?CombinationId $giftCombinationId,
+        bool $highlight,
+        bool $enabled,
+        ?DateTime $dateAdd,
+        ?DateTime $dateUpd
```

Comment:
Idk if we need to shorten this. It could be just $dateUpdated or even $updatedAt/$addedAt

Reply:
If not too big, it's better not to shorten variable names. The risk is very low but what could happen is someone mistakes this variable for something else (let's say he has a carrier called UPD, then he might think this is the timestamp of UPD delivery status) and start putting in it other kind of data. The clearer the better.

Reply:
Hmm i left them this way because in database they are date_add and date_upd. But of course I can change variable names here.